blackduck-artifactory-common/src/test/kotlin/com/synopsys/integration/blackduck/artifactory/modules/cancel/ScanCancelDeciderTest.java

```diff
@@ -0,0 +1,81 @@
+package com.synopsys.integration.blackduck.artifactory.modules.cancel;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.artifactory.fs.ItemInfo;
+import org.artifactory.repo.RepoPath;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.util.StringUtils;
+import org.mockito.Mockito;
+
+import com.synopsys.integration.blackduck.artifactory.ArtifactoryPAPIService;
+import com.synopsys.integration.blackduck.artifactory.PluginRepoPathFactory;
+import com.synopsys.integration.blackduck.artifactory.modules.scan.ScanModuleConfig;
+import com.synopsys.integration.blackduck.artifactory.modules.scan.ScanPropertyService;
+import com.synopsys.integration.blackduck.codelocation.Result;
+
+class ScanCancelDeciderTest {
+    @Test
+    void getCancelDecision() {
+        List<Result> scanResults = new ArrayList<>(Arrays.asList(Result.values()));
+        scanResults.add(null);
+
+        for (Result scanResult : scanResults) {
```

Comment:
Unless you have a compelling reason not to, I suggest using JUnit's @ParameterizedTest and @MethodSource annotations here to perform these tests.